src/Set.js

```diff
@@ -200,7 +206,7 @@ function updateSet(set, newMap) {
     set._map = newMap;
     return set;
   }
-  return newMap === set._map
+  return newMap.equals(set._map)
```

Comment:
For performance reasons, it's important that this is a cheap reference equality check. Specifically, we expect `newMap` to be a new reference if it has been altered

Reply:
Gotcha. I'll see what I can do to maintain the same reference in the map. I did notice that `map` and `mapEntries` have different code paths. Is there a recommendation on how I should approach this?

Reply:
I tested two approaches:
- doing `mapEntries` and keeping track if there was any alterations and only `updateSet` if changed
- using `withMutations` and `remove`/`add` when there is a change

The latter was faster in some micro-benchmarks so I've opted for that solution.